src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -86,14 +117,22 @@ public bool IsProcessingEnumerable()
         public void Reset()
         {
             AddMethodDelegate = null;
+            ConstructorParameterIndex = 0;
```

Comment:
This is a lot of extra state that is not obvious it is tied to ctor parameters. Can we either prefix these with "Ctor_" or something similar, or perhaps move to a new "ConstructorParameters" struct and reference that struct from here?

Reply:
I've moved them to a new struct.